modules/event-emitter/main/index.coffee

```diff
@@ -130,10 +134,22 @@ class EventEmitter
   # deregisters a callback
   off: (name, namespace, callback) ->
     if hx.isString(namespace)
-      @emittersMap.get(namespace)?.off(name, callback)
+      if @emittersMap.has(namespace)
+        be = @emittersMap.get(namespace)
+        be.off(name, callback)
+        if be.isEmpty()
+          removeEmitter(this, be, namespace)
     else
-      for emitter in @emitters.entries()
+      if not callback and not hx.isString(namespace)
+        callback = namespace
```

Comment:
Previously, this wasn't working correctly when using event emitters directly as you _had_ to supply a namespace to clear a specific callback.
Selection was working fine because we always pass all 3 arguments through.